python_modules/libraries/dagster-dg/dagster_dg/mcp/server.py

```diff
@@ -5,126 +5,242 @@
     """
-    return subprocess.check_output(
-        [
-            "uv",
-            "run",
-            "dg",
-            "list",
-            "plugins",
-            "--feature",
-            "component",
-        ],
-        cwd=project_path,
-        stderr=subprocess.STDOUT,
-    ).decode("utf-8")
+    try:
+        return subprocess.check_output(
+            [
+                "uv",
+                "run",
+                "dg",
+                "list",
+                "plugins",
+                "--feature",
+                "component",
+                "--json",
+            ],
+            cwd=project_path,
+            stderr=subprocess.STDOUT,
+        ).decode("utf-8")
+    except subprocess.CalledProcessError as e:
+        raise Exception("Subprocess failed", e.output)
 
 
 @mcp.tool()
 async def install_component(project_path: str, package_name: str) -> str:
     """Install a component from the marketplace.
-    Currently available components include:
+
+    Some available components include:
         - `dagster-dbt`
         - `dagster-sling`
+        - `dagster-evidence`
+
     Args:
-        project_path: The path to your Dagster project.
+        project_path: The full path to your Dagster project.
         package_name: The Python package available on pypi for the component.
 
     Returns:
         The output of the `uv add` command
     """
     # TODO - get list of available components from the registry
-    return subprocess.check_output(
-        ["uv", "add", package_name],
-        cwd=project_path,
-        stderr=subprocess.STDOUT,
-    ).decode("utf-8")
+    try:
+        return subprocess.check_output(
+            ["uv", "add", package_name],
+            cwd=project_path,
+            stderr=subprocess.STDOUT,
+        ).decode("utf-8")
+    except subprocess.CalledProcessError as e:
+        raise Exception("Subprocess failed", e.output)
 
 
 @mcp.tool()
 async def scaffold_dagster_component(
-    project_path: str, component_type: str, component_name: str
+    project_path: str,
+    component_type: str,
+    component_name: str,
 ) -> str:
     """Scaffold a new Dagster component in the project.
+
     This produces a component.yaml file which can be populated with the appropriate fields for the component.
 
     Args:
-        project_path: The path to your Dagster project.
-        component_type: The type of component to be scaffolded.
+        project_path: The full path to your Dagster project.
+        component_type: The full identifier of the component to be scaffolded (e.g. dagster_sling.SlingReplicationCollectionComponent)
+        component_name: The name of the component to provide for the newly scaffolded component.
+
+    Returns:
+        The output from running the `dg scaffold` command.
+    """
+    # NOTE: attempted to use `*args` and optional parameter without success
+    command = ["uv", "run", "dg", "--verbose", "scaffold", component_type, component_name]
```

Comment:
Tried using `*args` and an optional `additional_args` to pass specific-flags for the component with no luck, so `scaffold_dagster_dbt_component` was temporarily created as a workaround.